### ISSUE TITLE
False positive exhaustiveness check

### ISSUE THREAD

**Issue by ice1000** [intellij-rust/intellij-rust#3776]
## Environment

* **Intellij-Rust plugin version:** ![Screenshot from 2019-04-30 12-05-43](https://user-images.githubusercontent.com/16398479/56976162-4f0f6a00-6b40-11e9-830f-323872b33b6f.png)
* **Rust toolchain version:** Unrelated
* **IDE name and version:** ![Screenshot from 2019-04-30 12-05-27](https://user-images.githubusercontent.com/16398479/56976182-5cc4ef80-6b40-11e9-90a7-1bcd53177d6c.png)
* **Operating system:** Ubuntu 18.04 (I believe it's unrelated)

## Problem description
Given this minimum reproducible code:

```rust
enum Marisa {
    Ora,
    Muda,
    Dura,
}

fn main() {
    let _ = match (Marisa::Dura, Some(1_919_810)) {
        (Marisa::Ora, None) | (Marisa::Dura, Some(_)) => 23,
        (Marisa::Muda, None) | (Marisa::Muda, Some(_)) => 233,
        (Marisa::Ora, Some(b)) => b,
        (Marisa::Dura, None) => 233,
    };
}
```

IntelliJ Rust lies that it's not exhaustive:

![Screenshot from 2019-04-30 12-07-04](https://user-images.githubusercontent.com/16398479/56976232-7d8d4500-6b40-11e9-868d-25002170aa0f.png)

**Comment by ice1000** [comment]
Notice: I can compile the code above and everyone can confirm it to be exhaustive by eye.
